apps/web/src/components/organization-select.tsx

```diff
@@ -42,31 +42,45 @@ import {
   FormMessage,
 } from "./ui/form";
 import { Input } from "./ui/input";
+import { useToast } from "./ui/use-toast";
+
+
+
 
-type OrganizationForm = Omit<InsertOrganization, "owner">;
 
 export default function OrganizationSelect() {
   const pathname = usePathname();
-
   const organizationFromPathname = decodeURIComponent(
     // @ts-expect-error Since we are taking something from a pathname there has to be something
     pathname.split("/").at(2),
   );
-
   const router = useRouter();
+  const { toast } = useToast();
 
   const [isModalOpen, setIsModalOpen] = useState(false);
 
-  const { data: organizations, refetch } = useQuery({
+  useEffect(() => {
+    refetch();
+  }, [pathname]);
+
+  const { data: organizations, refetch, isLoading } = useQuery({
     queryKey: ["organizations"],
-    queryFn: () => getOwnOrganizations(),
+    queryFn: async () => {
+      const [organizations, err] = await getOwnOrganizations();
+
+      if (err !== null) {
+        console.error(err);
+      }
+
+      return organizations;
+    }
   });
 
-  function selectOrganization(orgId: string) {
-    router.push(`/dashboard/${orgId}/`);
+  function selectOrganization(orgName: string) {
+    router.push(`/dashboard/${orgName}/overview`);
   }
```

Comment:
Wouldn't wrapping the select options with `<Link />` make more sense?

Reply:
Since this function need to take the orgName dynamically it wouldn't make much sense, but I simplified this to:
`onValueChange={(newValue) => {
            router.push(`/dashboard/${newValue}/overview`);
          }}`